sinks/sink-mongo/tests/test_multi_collection.rs

```diff
@@ -0,0 +1,711 @@
+use apibara_core::node::v1alpha2::{Cursor, DataFinality};
+use apibara_sink_common::{Context, CursorAction, Sink};
+use apibara_sink_mongo::{MongoSink, SinkMongoError, SinkMongoOptions};
+use error_stack::{Result, ResultExt};
+use futures_util::TryStreamExt;
+use mongodb::{
+    bson::{doc, to_document, Bson, Document},
+    options::FindOptions,
+    Collection,
+};
+use serde_json::{json, Value};
+use testcontainers::{clients, core::WaitFor, GenericImage};
+
+fn new_mongo_image() -> GenericImage {
+    GenericImage::new("mongo", "7.0.1")
+        .with_wait_for(WaitFor::message_on_stdout("Waiting for connections"))
+}
+
+fn new_cursor(order_key: u64) -> Cursor {
+    Cursor {
+        order_key,
+        unique_key: order_key.to_be_bytes().to_vec(),
+    }
+}
```

Comment:
These 2 should go in a common module and shared between tests. See `sinks/sink-postgres/tests` to see what I mean.